src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationTracker.cs

```diff
@@ -170,7 +170,7 @@ public CompilationTracker FreezePartialStateWithTree(SolutionState solution, Doc
                     {
                         inProgressCompilation = inProgressCompilation.AddSyntaxTrees(tree);
```

Comment:
can you explain what this does? asking since I have some watson I am working on that looks like around our partial solution.

Reply:
It's doing something around partial solutions: a partial solution always must include a given document (the one you're in) -- this is forcing that in somehow.